cablelabs/transparent-security
# Arp multicast support

#### What does this PR do?
Fixes #303 and Fixes #316 
Implement ARP digest for insertion of L2 routes to the NB nodes
Removed the manual table entries into data forward from the lab trial tests.
Removed the addition of the switch_ethernet (-s) option denoting the dst_mac value in send_packets.py so the host generating the packets will leverage the dst_mac as in the ARP cache only for the lab_trial scenario tests.
Each NB node has ARP cache entries for all SB nodes and vice versa.

Not certifiable, but I was able to perform a curl call from host1 to inet with nginx running there on docker.
#### Do you have any concerns with this PR?
arp table on SB host is not getting the mac of the remote host so there could still be issues to be addressed in ticket #301 therefore currently the necessary arp cache entries are getting manually added but the P4 paths are still being learned only in the lab_trial scenario.
#### How can the reviewer verify this PR?
Check CI
Can also try pinging from host1 or host2 to inet in your own EC2 environment after the tests have completed.
If you are really adventurous, you can fire up nginx on inet (docker installed not running) and see if you can connect from host1 or host2; however, if that does not work, I will be addressing those issues next.
#### Any background context you want to provide?
A step towards TCP support on Tofino
#### Screenshots or logs (if appropriate)
#### Questions:
- Have you connected this PR to the issue it resolves? yes
- Does the documentation need an update? no
- Does this add new dependencies? no
- Have you added unit or functional tests for this PR? updated the existing ones
